xbmc/pvr/guilib/PVRGUIActionsPlayback.cpp

```diff
@@ -190,9 +203,65 @@ bool CPVRGUIActionsPlayback::PlayRecording(const CFileItem& item, bool bCheckRes
 
   if (!bCheckResume || CheckResumeRecording(item))
   {
-    CFileItem* itemToPlay = new CFileItem(recording);
-    itemToPlay->SetStartOffset(item.GetStartOffset());
-    StartPlayback(itemToPlay, true);
+    if (!item.m_bIsFolder && VIDEO_UTILS::IsAutoPlayNextItem(item))
+    {
+      // recursively add items located in the same folder as item to play list, starting with item
+      std::string parentPath = item.GetProperty("ParentPath").asString();
+      if (parentPath.empty())
+        URIUtils::GetParentPath(item.GetPath(), parentPath);
+
+      if (parentPath.empty())
+      {
+        CLog::LogF(LOGERROR, "Unable to obtain parent path for '{}'", item.GetPath());
+        return false;
+      }
+
+      const auto parentItem = std::make_shared<CFileItem>(parentPath, true);
+      if (item.GetStartOffset() == STARTOFFSET_RESUME)
+        parentItem->SetStartOffset(STARTOFFSET_RESUME);
+
+      auto queuedItems = std::make_unique<CFileItemList>();
+      VIDEO_UTILS::GetItemsForPlayList(parentItem, *queuedItems);
+
+      // figure out where to start playback
+      int pos = 0;
+      for (const auto& i : *queuedItems)
```

Comment:
maybe avoid `i` and `auto` here to increase readability. i seems to be actually a fileitem not some index value :)

Reply:
Yep. I will change that.